docs/design/hts_address_tx_receipt.md

```diff
@@ -0,0 +1,62 @@
+# HTS Token Creation via SystemContract Receipt Enhancement
+
+## Purpose
+The purpose of this enhancement is to improve the interoperability between Hedera Token Service (HTS) and Ethereum-compatible tools, like ethers, by ensuring access to the necessary information about newly created tokens, in this case token address. This change specifically addresses the gap in Web3 tooling support when creating tokens through HTS precompile contracts, making user experience smoother.
+
+
+## Problem Statement
+When users create tokens (fungible and non-fungible) by directly calling the 0x167 contract there is no way for them to get the evm address of the newly created token via standard EVM tooling e.g ethers. This is because the transaction receipt's contractAddress field is not being populated with the HTS token address, since essentially the operations is just a contract call, but not a contract create (deploy).
+
+## Current Behaviour
+
+Currently, when creating a token via HTS the address of the system contract is returned as a contractAddress in the transaction receipt.
+
+## Proposed Solution
+
+### Technical Details
+
+### 1. Function Signature Detection
+
+Detect in the transactionResponse from the mirror node, if the transaction was calling any of the HTS methods - createFungibleToken/createNonFungibleToken/createFungibleTokenWithCustomFees/createNonFungibleTokenWithCustomFees
```

Comment:
`if the transaction was calling any of the HTS methods` this could be a little misleading as the listed methods are only about token creation. 
```suggestion
Detect in the transactionResponse from the mirror node, if the transaction was calling any of the token creation HTS methods - createFungibleToken/createNonFungibleToken/createFungibleTokenWithCustomFees/createNonFungibleTokenWithCustomFees
```

Reply:
I think the purpose of this sentence is to name the specific methods in the end.
If you really want it to be clear i can change it to
-
```
Detect in the transactionResponse from the mirror node, if the transaction was calling any of the FOLLOWING HTS methods - createFungibleToken/createNonFungibleToken/createFungibleTokenWithCustomFees/createNonFungibleTokenWithCustomFees
```
As Nana mentioned in another comment this may change to include other methods as well, so i am reluctant to specifying if they are token creation or anything else. What do you think

Reply:
Nice! I like it! Thanks for the update!